src/core/pointcloud/qgscopcpointcloudindex.cpp

```diff
@@ -157,6 +157,11 @@ QgsPointCloudBlock *QgsCopcPointCloudIndex::nodeData( const IndexedPointCloudNod
   std::ifstream file( QgsLazDecoder::toNativePath( mFileName ), std::ios::binary );
   file.seekg( blockOffset );
   file.read( rawBlockData.data(), blockSize );
+  if ( !file )
```

Comment:
why is this check after the seek/read?

Reply:
Both `seekg()` and `read()` may set the `badbit` in `ifstream`, so I thought we can also handle some unexpected `ifstream` error (eg blockOffset out of bounds?)

Reply:
Ok, sounds fair enough to me!